rs/p2p/consensus_manager/src/sender.rs

```diff
@@ -238,76 +234,82 @@ impl<
+                    if !is_initiated {
+                        let child_token = cancellation_token.child_token();
+                        let child_token_clone = child_token.clone();
+                        metrics.send_view_send_to_peer_total.inc();
 
-                            let transport = transport.clone();
-                            let body = body.clone();
+                        let transport = transport.clone();
+                        let body = body.clone();
+                        let route = route.clone();
 
-                            let send_future = async move {
-                                select! {
-                                    _ = send_advert_to_peer(transport, body, peer, uri_prefix::<WireArtifact>()) => {},
-                                    _ = child_token.cancelled() => {},
-                                }
-                            };
+                        let send_future = async move {
+                            select! {
+                                _ = send_transmit_to_peer(transport, body, peer, route) => {},
+                                _ = child_token.cancelled() => {},
+                            }
+                        };
 
-                            in_progress_transmissions.spawn_on(send_future, &rt_handle);
-                            initiated_transmissions.insert(peer, (connection_id, child_token_clone));
-                        }
+                        in_progress_transmissions.spawn_on(send_future, &rt_handle);
+                        initiated_transmissions.insert(peer, (connection_id, child_token_clone));
                     }
                 }
-                Some(result) = in_progress_transmissions.join_next() => {
+            }
+            Some(result) = in_progress_transmissions.join_next() => {
+                panic_on_join_err(result);
+                metrics.send_view_send_to_peer_delivered_total.inc();
+            }
+            _ = cancellation_token.cancelled() => {
+                while let Some(result) = in_progress_transmissions.join_next().await {
+                    metrics.send_view_send_to_peer_cancelled_total.inc();
                     panic_on_join_err(result);
-                    metrics.send_view_send_to_peer_delivered_total.inc();
-                }
-                _ = cancellation_token.cancelled() => {
-                    while let Some(result) = in_progress_transmissions.join_next().await {
-                        metrics.send_view_send_to_peer_cancelled_total.inc();
-                        panic_on_join_err(result);
-                    }
-                    break;
                 }
+                break;
             }
         }
     }
 }
 
-/// Sends a serialized advert or artifact message to a peer.
-/// If the peer is not reachable, it will retry with an exponential backoff.
 #[instrument(skip(transport, message))]
-async fn send_advert_to_peer(
+async fn send_transmit_to_peer(
```

Comment:
Aren't send and transmit synonyms? If you want to remove the `advert` term, wouldn't `message` be a better replacement? I.e. `send_message_to_peer`.

Reply:
I started using transmit as the logic unit in the consensus manager, this is similar naming to quinn